TeamSnapSDK/SDK/DataTypes/Messages/TSDKMessageDatum.m

```diff
@@ -8,9 +8,15 @@
 
 #import "TSDKMessageDatum.h"
 
+@interface TSDKMessageDatum ()
+
+@property (nonatomic, weak) NSString *_Nullable messageType;
```

Comment:
What if we added a custom setter for `setMessageType:(NSString *)messageType` that auto-mapped to our new typedef'd type? This would obviate the need to have `messageTypeOfUnreadCount` doing string comparison at read-time. It would only do it once when the property was written.